### PR TITLE
OSDOCS-9435: configuring router admission policy

### PR DESCRIPTION
**Pull request by aravipra**
Version(s):
4.16+

Issue:
[OSDOCS-9435](https://issues.redhat.com/browse/OSDOCS-9435)

Link to docs preview:
[Configuring route admission policy](https://76317--ocpdocs-pr.netlify.app/microshift/latest/microshift_networking/microshift-nw-router.html)

QE review:
- [x] QE has approved this change.

SME review:
- [x] SME has approved this change.

Additional information:
 [Releated feature PR](https://github.com/openshift/openshift-docs/pull/76100)
 [release notes of this PR](https://github.com/openshift/openshift-docs/pull/76320)

### CODE REVIEW

File: modules/microshift-nw-config-router-admission.adoc
Comment (pacevedom) [inline modules/microshift-nw-config-router-admission.adoc:18]: I think we should mention the other value for this too, which is `Strict`.

File: modules/microshift-nw-config-router-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-router-admission.adoc:25]: I think it will make sense to keep this. It gives continuity to users who refer this section. WDYT?

File: modules/microshift-nw-config-router-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-router-admission.adoc:18]: The procedure explains about the route configuration option to disable the checks which supports combining routes from multiple namespaces.
How about I  add a note below about `strict` ?

File: modules/microshift-nw-config-route-admission.adoc
Comment (pacevedom) [inline modules/microshift-nw-config-route-admission.adoc:24]: Could we lose the apiServer part?

**Review by pacevedom — COMMENTED** [review]
just one minor nit, otherwise lgtm

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:7]:
```suggestion
= Configuring the route admission policy
```

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:18]:
```suggestion
.  To prevent routes in different namespaces from claiming the same hostname, update the `namespaceOwnership` field value to `Strict` in the {microshift-short} `config.yaml` file as shown in the following example:
```
* The _parameter_ is the "key" part of the YAML, the thing left of the colon, so in this case, `ingress.routeAdmissionPolicy.namespaceOwnership` (or just `namespaceOwnership` since we do give the stanza here). The _value_ of the field the parameter defines is what we are changing.

* We want to keep "in the following example" as the last thing before the example and move the clause earlier in the sentence. It's more user-focused and is important for accessibility (think screen readers). 

* It is worthwhile to ask @ShudiLi what happens if a user does delete the whole `routeAdmissionPolicy.namespaceOwnership` parameter. We have some funky behavior with other parameters based on whether they are deleted or left empty.

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:9]:
```suggestion
By default, {microshift-short} allows routes in multiple namespaces to use the same hostname. To prevent routes in different namespaces from claiming the same hostname, you can configure the value. 
```
It defaults to allowing. See https://github.com/openshift/openshift-docs/pull/76100/files#diff-5968debd1075f10b57e348a65f0cedec425dc131c575c44f417d5ebe0a272841R76 per QE comments.

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:20]:
```suggestion
.Example `config.yaml` route admission policy
```
We're setting a single policy, so I think it's OK to shorten for the sake of minimalism.

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:26]:
```suggestion
    namespaceOwnership: Strict <1>
```
We're giving an example config, not the default.

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:29]:
```suggestion
<1> Valid values are `Strict` and `InterNamespaceAllowed`. Specifying `Strict` prevents routes in different namespaces from claiming the same host. If the value is deleted in a customized {microshift-short} `config.yaml`, the `InterNamespaceAllowed` value is automatically set.
```

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:30]:
```suggestion
```
This can be deleted. The checking is a backend/implementation behavior and I don't think it's relevant. The user side is that we're restricting or allowing access.

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:31]:
```suggestion
```

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:43]: Is there a simple Verification step we should add somewhere?

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:23]:
```suggestion
# ....
```

File: microshift_networking/microshift-nw-router.adoc
Comment (ShaunaDiaz) [inline microshift_networking/microshift-nw-router.adoc:9]:
```suggestion
Learn about default and custom settings for configuring the router and route admission policy with {microshift-short}.
```
I would add this to the assembly, too.

File: modules/microshift-nw-config-route-admission.adoc
Comment (pacevedom) [reply modules/microshift-nw-config-route-admission.adoc:9]: This was my bad when reading it, by default it *allows* routes. This is contrary to what OpenShift does, which defaults to Strict.

Comment (ShaunaDiaz) [reply modules/microshift-nw-config-route-admission.adoc:9]: So please make all of the changes I suggested here, @aravipra.

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [reply modules/microshift-nw-config-route-admission.adoc:30]: @pacevedom Am I right on this? WDYT? Is it important for end users to know the system behavior, or just how to configure the `namespaceOwnership` value?

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:7]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:13]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:18]: updated suggestions. 
Checking with @ShudiLi

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:9]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:20]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:26]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:31]: This is continuation of the above line.

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:43]: @pacevedom we had discussed to omit verification. Should I add this or not? Please suggest

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:23]: done

File: microshift_networking/microshift-nw-router.adoc
Comment (aravipra) [reply microshift_networking/microshift-nw-router.adoc:9]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:30]:
```suggestion
```
delete this sentence

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [reply modules/microshift-nw-config-route-admission.adoc:42]: We removed them from the other PR. MicroShift does show the current config when it starts anyway, and we have the `show config` in nearby docs. Sorry for the noise, disregard.

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:30]: done

**Comment by ShudiLi** [comment]
/lgtm
thanks

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [inline modules/microshift-nw-config-route-admission.adoc:23]:
```suggestion
# ...
```
Per peer review I received on the other PR. (it's three dots)

Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:23]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:27]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (xenolinux) [inline modules/microshift-nw-config-route-admission.adoc:18]:
```suggestion
. To prevent routes in different namespaces from claiming the same hostname, update the `namespaceOwnership` field value to `Strict` in the `config.yaml` file. See the following example:
```

File: modules/microshift-nw-config-route-admission.adoc
Comment (xenolinux) [inline modules/microshift-nw-config-route-admission.adoc:9]:
```suggestion
By default, {microshift-short} allows routes in multiple namespaces to use the same hostname. You can prevent routes from claiming the same hostname in different namespaces by configuring the route admission policy.
```

File: modules/microshift-nw-config-route-admission.adoc
Comment (xenolinux) [reply modules/microshift-nw-config-route-admission.adoc:14]: Suggestion to maintain the same format as other prereq and to avoid a passive voice.

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShudiLi) [reply modules/microshift-nw-config-route-admission.adoc]: @ShaunaDiaz If the whole `routeAdmissionPolicy.namespaceOwnership` parameter is delete, the default value `InterNamespaceAllowed` will be set to `routeAdmissionPolicy.namespaceOwnership`. If there isn't the config.yaml file, all parameters are set to the default values(If there is a parameter in the config.yaml file, the default value will be set to it)

File: modules/microshift-nw-config-route-admission.adoc
Comment (xenolinux) [reply modules/microshift-nw-config-route-admission.adoc:29]: IMO, a general flow should be: Description of a field --> valid values --> more information.

File: modules/microshift-nw-config-route-admission.adoc
Comment (xenolinux) [inline modules/microshift-nw-config-route-admission.adoc:33]:
```suggestion
If you complete the `config.yaml` file configuration at the same time, you can minimize system restarts.
```

File: modules/microshift-nw-config-route-admission.adoc
Comment (xenolinux) [inline modules/microshift-nw-config-route-admission.adoc:36]: Can we possibly provide a short reason why restart the service? 
```suggestion
. To apply the configuration, restart the {microshift-short} service by running the following command:
```

**Review by xenolinux — COMMENTED** [review]
A few suggestions to consider, otherwise LGTM!

/remove-label peer-review-needed
/remove-label peer-review-in-progress
/label peer-review-done

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [reply modules/microshift-nw-config-route-admission.adoc:18]: Yes, it is required to mention the specific file here.

**Comment by ShaunaDiaz** [comment]
@aravipra Be certain to rebase this PR and pick up all the merged changes to this assembly.

**Comment by ShaunaDiaz** [comment]
@aravipra Also get SME approval from @jerpeter1 before proceeding to merge review.

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:14]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:29]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:18]: second suggestion is accepted.

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:9]: done

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:33]: Yes, that is the meaning. But I am retaining this Tip as is to maintain consistency with this https://github.com/openshift/openshift-docs/pull/76100 PR.

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:36]: I am retaining this Tip as is to maintain consistency with this https://github.com/openshift/openshift-docs/pull/76100 PR.

File: modules/microshift-nw-config-route-admission.adoc
Comment (xenolinux) [reply modules/microshift-nw-config-route-admission.adoc:34]: Would you consider rephrasing these `[TIP]` instances in a separate follow up PR (or in this PR)? Because, right now given the context of this section, the tip is not meaningful. 
User might miss this important detail if not rephrased.

Comment (xenolinux) [reply modules/microshift-nw-config-route-admission.adoc:34]: You can maintain the consistency by rephrasing all similar occurrences of TIP.

File: modules/microshift-nw-config-route-admission.adoc
Comment (xenolinux) [reply modules/microshift-nw-config-route-admission.adoc:37]: Same comment https://github.com/openshift/openshift-docs/pull/76317#discussion_r1625204882

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:34]: As per our discussion retaining this TIP as is.

File: modules/microshift-nw-config-route-admission.adoc
Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:37]: made this change. done.

**Comment by ShaunaDiaz** [comment]
@aravipra [This module](https://github.com/openshift/openshift-docs/pull/76317/files#diff-341b40a7aa13fe87b04949ca7a0ea971773e8bf15fef4e7794fff0169639fb3fR6) should come after the others, OR you can change the abstract sentence [here](https://github.com/openshift/openshift-docs/pull/76317/files#diff-341b40a7aa13fe87b04949ca7a0ea971773e8bf15fef4e7794fff0169639fb3fR6) and flip the words around (e.g, Learn about default and custom settings for configuring the route admission policy and router with {microshift-short}). Yeah, we should update the [assembly title here](https://github.com/openshift/openshift-docs/pull/76317/files#diff-f1b8e8add585faba312bef818044083e5bb0d0788d473bfd68852c6411dd5666R3) to Understanding and configuring ingress. I don't think it matters which order these things are in as long as it is consistent.

File: modules/microshift-nw-config-route-admission.adoc
Comment (ShaunaDiaz) [reply modules/microshift-nw-config-route-admission.adoc:34]: We could do something like:
You can minimize system restarts by planning the configuration needs for your use case in advance and then by making all of the necessary changes to the {microshift-short} `config.yaml` file at the same time.

Comment (aravipra) [reply modules/microshift-nw-config-route-admission.adoc:34]: making this change as per discussion with Shauna.

**Review by skopacz1 — APPROVED** [review]
Merge review LGTM

**Pull request merged**
